mino/router/tree/tree.go

```diff
@@ -98,24 +114,28 @@ func (t *dynTree) GetMaxHeight() int {
 
 // GetRoute implements tree.Tree. It returns the address to route the target, or
 // nil if no route is found.
-func (t *dynTree) GetRoute(to mino.Address) mino.Address {
+func (t *dynTree) GetRoute(to mino.Address) (mino.Address, error) {
 	t.Lock()
 	defer t.Unlock()
 
+	if t.offline.Search(to) {
```

Comment:
so, once a node is offline, it never gets a chance to be brought back in the tree?

Reply:
Yes, for the simple tree router. Note that this is for a given protocol. It does not persist state between protocol executions. So next calls will try again to contact the node.